### PR TITLE
invidious: 0.20.1-unstable-2024-04-10 -> 2.20240427

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelog: https://github.com/iv-org/invidious/releases/tag/v2.20240427

Invidious has now switched to a conventional release model.
Thus, [our update script](https://github.com/NixOS/nixpkgs/blob/2b1f64b358f2cab62617f26b3870fd0ee375d848/pkgs/servers/invidious/update.sh) has to be adapted in consequence. I would be glad to get some help on this.

cc @infinisil @jbedo @999eagle 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by 999eagle** [comment]
to ease using the update script i think we should just stick to the existing `versions.json` to make things a bit easier on us. as i can't push to this branch, i've updated the update script and fixed the invidious version string here: in the meantime: https://github.com/GaetanLepage/nixpkgs/compare/invidious...999eagle:nixpkgs:invidious

the last commit on there might be out of scope for this pr but it would allow easily overriding the version of invidious to build which i feel would be good for allowing people to continue tracking `master` directly

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 307083` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>invidious</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 307083` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>invidious</li>
  </ul>
</details>

File: pkgs/servers/invidious/default.nix [hunk 1]
```diff
@@ -23,7 +34,8 @@ crystal.buildCrystalPackage rec {
     owner = "iv-org";
     repo = pname;
     fetchSubmodules = true;
-    inherit (versions.invidious) rev hash;
+    rev = if versions.invidious ? rev then versions.invidious.rev else "v${version}";
```
Comment (GaetanLepage) [inline pkgs/servers/invidious/default.nix:37]:
```suggestion
    rev = if versions.invidious.rev or "v${version}";
```
Couldn't this work ?

Comment (999eagle) [reply pkgs/servers/invidious/default.nix:37]: oh yeah that'd work too, thanks! feel free to apply that patch yourself as it's on your branch anyway :)

Comment (GaetanLepage) [reply pkgs/servers/invidious/default.nix]: Applied the patch !

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 307083` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>invidious</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 307083` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>invidious</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
@infinisil I think that this should be ready to be merged.
Knowing that invidious is currently broken (videos are not playable at all), shipping this update is a bit more urgent than the average ones.

Ofc, if you do not have the time for this, we will try to find someone else to review it.

Take care !

File: pkgs/servers/invidious/default.nix [hunk 2]
```diff
@@ -110,6 +126,10 @@ crystal.buildCrystalPackage rec {
     mainProgram = "invidious";
     homepage = "https://invidious.io/";
     license = licenses.agpl3Plus;
-    maintainers = with maintainers; [ sbruder ];
+    maintainers = with maintainers; [
+      _999eagle
+      GaetanLepage
+      sbruder
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/servers/invidious/default.nix:132]: Done

File: pkgs/servers/invidious/update.sh [hunk 3]
```diff
@@ -37,38 +36,34 @@ if [ ! -d "$git_dir" ]; then
...
 
-if [ "$new_rev" = "$old_rev" ]; then
+if [ "$new_version" = "$old_version" ]; then
     info "$pkg is up-to-date."
     exit
 fi
 
+commit="$(git -C "$git_dir" rev-list "$new_tag" --max-count=1 --abbrev-commit)"
+date="$(git -C "$git_dir" log -1 --format=%cd --date=format:%Y.%m.%d)"
+json_set '.invidious.version_string' "$date-$commit"
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/servers/invidious/update.sh]: Good idea indeed. Done.

**Review by 999eagle — APPROVED** [review]
thanks for the fixes! lgtm!

**Comment by pbsds** [comment]
the darwin builders seem sad

@ofborg build invidious

**Review by pbsds — APPROVED** [review]

**Comment by GaetanLepage** [comment]
> the darwin builders seem sad

The darwin platform is not supported for this package.

**Pull request merged**